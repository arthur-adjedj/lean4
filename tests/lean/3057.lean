mutual
  inductive Tree : Type :=
    | node : ListTree → Tree
    deriving Repr, DecidableEq, BEq, Hashable, Ord

  inductive ListTree : Type :=
    | nil : ListTree
    | cons : Tree → ListTree → ListTree
    deriving Repr, DecidableEq, BEq, Hashable, Ord
end

#synth Repr Tree
#synth Repr ListTree
#synth DecidableEq Tree
#synth DecidableEq ListTree
#synth BEq Tree
#synth BEq ListTree
#synth Hashable Tree
#synth Hashable ListTree
#synth Ord Tree
#synth Ord ListTree
