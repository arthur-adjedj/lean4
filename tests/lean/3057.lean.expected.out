instReprTree_1
instReprListTree_1
instDecidableEqTree_1
instDecidableEqListTree_1
instBEqTree_1
instBEqListTree_1
instHashableTree_1
instHashableListTree_1
instOrdTree_1
instOrdListTree_1
